test/e2e/test_guest_user_reads.py

```diff
@@ -63,10 +81,23 @@ def test_guest_user_private_dataset_read(
 
         assert "not found" in str(error)
 
+        # Logged data should also not be readable.
+        dataset = client.data_catalog.get_dataset_by_name(
+            populated_project.id, "dataset1"
+        )
+
+        with pytest.raises(LayerClientException) as error:
```

Comment:
there is a dedicated exception for this, should be sthg like `LayerClientResourceNotFound`